File: ecolink-front/src/app/core/services/recycling-point.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RecyclingPoint } from '../models/index';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class RecyclingPointService {
  private base = `${environment.apiUrl}/recycling-points`;
  constructor(private http: HttpClient) {}

  getAllActive(): Observable<RecyclingPoint[]> {
    return this.http.get<RecyclingPoint[]>(this.base);
  }
  getById(id: number): Observable<RecyclingPoint> {
    return this.http.get<RecyclingPoint>(`${this.base}/${id}`);
  }
  create(point: Partial<RecyclingPoint>): Observable<RecyclingPoint> {
    return this.http.post<RecyclingPoint>(this.base, point);
  }
  update(id: number, point: Partial<RecyclingPoint>): Observable<RecyclingPoint> {
    return this.http.put<RecyclingPoint>(`${this.base}/${id}`, point);
  }
  getNearby(latitude: number, longitude: number, radiusKm: number = 5): Observable<RecyclingPoint[]> {
    const params = new HttpParams()
      .set('latitude', latitude)
      .set('longitude', longitude)
      .set('radiusKm', radiusKm);
    return this.http.get<RecyclingPoint[]>(`${this.base}/nearby`, { params });
  }
}

File: flutter_application_1/lib/services/recycling_point_service.dart
import '../models/recycling_point.dart';
import 'api_client.dart';

class RecyclingPointService {
  static Future<List<RecyclingPoint>> getAllActive() async {
    final json = await ApiClient.get('/recycling-points');
    return (json as List? ?? [])
        .map((e) => RecyclingPoint.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  static Future<List<RecyclingPoint>> getNearby(
      double latitude, double longitude, {double radiusKm = 5.0}) async {
    final json = await ApiClient.get('/recycling-points/nearby', query: {
      'latitude': latitude,
      'longitude': longitude,
      'radiusKm': radiusKm,
    });
    return (json as List? ?? [])
        .map((e) => RecyclingPoint.fromJson(e as Map<String, dynamic>))
        .toList();
  }
}
